packages/core/lib/commands/test/run.js

```diff
@@ -3,13 +3,27 @@ const parseCommandLineFlags = options => {
   const grep = options.grep || options.g;
   const bail = options.bail || options.b;
   const reporter = options.reporter || options.r;
-  return {
-    mocha: {
-      grep,
-      bail,
-      reporter
-    }
-  };
+
+  /**
+   * This if-else condition is explicitly defined so that when the reporter is not specified by the user
```

Comment:
This comment doesn't really explain the reasoning, I'm afraid.  It's not the sort of comment that would stop a well-meaning simplifier from taking out the conditional.

The point of comments is to point out things that are non-obvious or counterintuitive, and what's counterintuitive about this code is the fact that `{ mocha, bail }` acts differently from `{ mocha, bail, reporter: undefined }`.  This is an uncommon situation in Javascript; typically, having a member equal to `undefined`, and having it not present, act the same.  Thus, a well-meaning simplifier who was not aware that the distinction was significant in this case, would likely remove this branch, re-introducing the problem.

I think the comment should make a point of expressing that the undefined-vs-absent distinction really is significant in this case.

Reply:
Thanks @haltman-at . I will make the changes.

Reply:
Hm, I don't see how the rewritten comment accomplishes the goal.  It doesn't highlight the distinction between absence and undefined.  Again, most readers will think of these as being generally the same, so when you say "If the reporter is returned as undefined", most readers will think of that as being the same as it being absent.  In order to highlight the distinction, you should add a clause like "as opposed to completely absent".  More generally, if you want to highlight a distinction, you need to present both its sides, not just one of them.